src/java.base/windows/classes/sun/nio/fs/WindowsNativeDispatcher.java

```diff
@@ -976,15 +976,22 @@ static void CreateSymbolicLink(String link, String target, int flags)
             CreateSymbolicLink0(linkBuffer.address(), targetBuffer.address(),
                                 flags);
         } catch (WindowsException x) {
+            // Retry if the privilege to create symbolic links is not held
             if (x.lastError() == ERROR_PRIVILEGE_NOT_HELD) {
                 flags |= SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE;
                 try {
                     CreateSymbolicLink0(linkBuffer.address(),
                                         targetBuffer.address(), flags);
                     return;
-                } catch (WindowsException ignored) {
-                    // Will fail with ERROR_INVALID_PARAMETER for Windows
-                    // builds older than 14972.
+                } catch (WindowsException y) {
+                    // Throw an exception if and only if it is not due to symbolic link creation
+                    // privilege not being held (ERROR_PRIVILEGE_NOT_HELD) nor the
+                    // SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE flag not being recognized
+                    // (ERROR_INVALID_PARAMETER). The latter will occur for Windows builds
+                    // older than 14972.
+                    int lastError = y.lastError();
+                    if (lastError != ERROR_PRIVILEGE_NOT_HELD && lastError != ERROR_INVALID_PARAMETER)
+                        throw y;
```

Comment:
The methods that WindowsNativeDispatcher defines are meant to be 1-1 mapping to win32 calls. It looks like we put the changes for JDK-8221852 in the wrong place, they need to be the usage rather than here.

Reply:
Restored `WindowsNativeDispatcher` 1:1 mapping for `CreateSymbolcLink`: logic moved to a new static method in `WindowsLinkSupport` (255f5c7f8d86a3d653fb648ed993bb4589792881).